### PR TITLE
Add subclass hierarchy test

### PR DESCRIPTION
**Pull request by shreyasnagare**
This test can help us find cycles in the subclass hierarchy.

### CODE REVIEW

**Comment by gtfierro** [comment]
Good find! I'll approve this as soon as the tests pass

File: tests/test_subclass_hierarchy.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import sys
+from rdflib import RDFS, Graph
+
+sys.path.append("..")
+from bricksrc.namespaces import BRICK
```
Comment (gtfierro) [inline tests/test_subclass_hierarchy.py]: This should have been flagged by the style checker. Can you make sure `pre-commit` is installed (see instructions @ https://github.com/BrickSchema/Brick/blob/master/CONTRIBUTING.md) and see if this gets flagged?

File: tests/test_subclass_hierarchy.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        "SELECT ?brick_class WHERE { ?brick_class rdfs:subClassOf+ ?brick_class .}"
+    )
+    for row in res1:
+        classes_in_cycles.append(row[0])
+
+    # Find all the triples involved.
+    loops = set()
+    for brick_class in classes_in_cycles:
+        res2 = g.query(
+            """SELECT DISTINCT ?s ?o WHERE {
```
Comment (gtfierro) [inline tests/test_subclass_hierarchy.py]: Stylistically, can you fold this string back to the line above?

```python
res2 = g.query("""SELECT # and so on
```

File: tests/test_subclass_hierarchy.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                              ?s rdfs:subClassOf ?o .
+                              brick:%s rdfs:subClassOf+ ?s .
+                              ?o rdfs:subClassOf+ brick:%s .
+                        }"""
+            % (minify(brick_class), minify(brick_class))
+        )
+        for s, o in res2:
+            loops.add("%s -> subClassOf -> %s" % (minify(s), minify(o)))
+    assert not loops, f"Loops found in the class hierarchy!"
```
Comment (gtfierro) [inline tests/test_subclass_hierarchy.py]: Can you make the test explicit? `assert len(loops) == 0` so its more obvious what the test is checking?

File: tests/test_subclass_hierarchy.py (hunk 2 above)
Comment (shreyasnagare) [reply tests/test_subclass_hierarchy.py]: `Black` formats it like that, I guess. Similar to queries on other files: https://github.com/BrickSchema/Brick/blob/f8c9a4d4ae2cb0206e8719b97b29245bab0239ec/tests/test_no_inference.py#L112

Comment (gtfierro) [reply tests/test_subclass_hierarchy.py]: Ok! Far be it from me to question Black's output :)

File: tests/test_subclass_hierarchy.py (hunk 1 above)
Comment (shreyasnagare) [reply tests/test_subclass_hierarchy.py]: You're right, I forgot to install pre-commit on my new clone.

**Comment by shreyasnagare** [comment]
It should pass all the tests once #114 is merged in.

**Comment by gtfierro** [comment]
One thing I'm realizing with my suggestion of `assert len(loops) == 0` is we don't get what the loops are. Can you add the loops themselves to the error message? Something like 

```python
     assert len(loops) == 0, f"Loops found in the class hierarchy for classes: {loops}"
```

Open to any ideas you have for how to make the loops themselves obvious

**Comment by gtfierro** [comment]
I merged in #114 so these tests should pass now

**Comment by shreyasnagare** [comment]
> One thing I'm realizing with my suggestion of `assert len(loops) == 0` is we don't get what the loops are. Can you add the loops themselves to the error message? Something like
> 
> ```python
>      assert len(loops) == 0, f"Loops found in the class hierarchy for classes: {loops}"
> ```
> 
> Open to any ideas you have for how to make the loops themselves obvious

I still think your suggestion is okay. When I try to add the triple `brick:HVAC rdfs:subClassOf brick:RTU .`, the test throws the following error:
```
E       AssertionError: Loops found in the class hierarchy!
E       assert 3 == 0
E        +  where 3 = len({'AHU -> subClassOf -> HVAC', 'HVAC -> subClassOf -> RTU', 'RTU -> subClassOf -> AHU'})
```

I think this output is good enough to identify the issue.

**Comment by jbkoh** [comment]
A great addition. LGTM. Thanks. I am merging this one.

**Pull request merged**